### ISSUE TITLE
`set_id` issue with included

### ISSUE THREAD

**Issue by ababich** [Netflix/fast_jsonapi#237]
Picture simple relationship like this:
```
class ASerializer
  include FastJsonapi::ObjectSerializer

  has_many :bs
end

class BSerializer
  include FastJsonapi::ObjectSerializer

  set_id :custom_id
end
```

The problem is that `fast_jsonapi`to serialize A objects with B included will generate structure like this:
```
{
  data: [
    { id: AID,
      type: :a,
      attributes: {},
      relationships: {
       bs: [
         { id: B1ID, type: :b },
         { id: B2ID, type: :b }
       ]       
      }
  ],
  included: [
    { id: B1CUSTOMID, ... },
    { id: B2CUSTOMID, ... },
  ]
}
```

So that unless we specifically set `id_method_name` in relationship and implement it accordingly this makes no sense

I suggest to add additional checks to prevent this situation, i.e. have `included` ids not matching `relationships` ids

----

This is backlog feature request, in general, as it improves API but not changes functionality and may affect performance

**Comment by shishirmk** [comment]
This is interesting. I believe this could be a bug. Do you mind writing a failing test to better capture this issue? I can try to fix it if you could do that.

**Comment by ababich** [comment]
Sure, please take a look at https://github.com/ababich/fast_jsonapi/compare/master...set_id_fail

**Comment by yale** [comment]
hi folks - just bumping this as we're running into this same issue as well.

**Comment by shishirmk** [comment]
@yale there is a workaround for this as @ababich mentioned you need to set both `id_method_name` and `set_id`. We would need to figure out how to `set_id` so that it can be used both in the serializer and all the places the record is used in a relationship

**Comment by shishirmk** [comment]
Released in [Version 1.4](https://github.com/Netflix/fast_jsonapi/releases/tag/1.4)

**Comment by zinosama** [comment]
@shishirmk I don't think this issue is resolved by #274 since I'm still running into the same problem with 1.4. As manojmj92 [pointed out](https://github.com/Netflix/fast_jsonapi/pull/274#issuecomment-406156094), #274 only solves the issue with `id_method_name` not being set with block.
